{
  service: 'user-service',
  message: "Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/",
  errorLabelSet: Set(0) {},
  reason: TopologyDescription {
    type: 'ReplicaSetNoPrimary',
    servers: Map(3) {
      'ac-hqc0sa6-shard-00-00.ep6vddc.mongodb.net:27017' => ServerDescription {
        address: 'ac-hqc0sa6-shard-00-00.ep6vddc.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        minRoundTripTime: 0,
        lastUpdateTime: 424325,
        lastWriteDate: 0,
        error: MongoNetworkError: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
        
            at TLSSocket.<anonymous> (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongodb\lib\cmap\connect.js:286:44)
            at Object.onceWrapper (node:events:633:26)
            at TLSSocket.emit (node:events:518:28)
            at emitErrorNT (node:internal/streams/destroy:170:8)
            at emitErrorCloseNT (node:internal/streams/destroy:129:3)
            at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
          errorLabelSet: Set(1) { 'ResetPool' },
          beforeHandshake: false,
          [cause]: [Error: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
          ] {
            library: 'SSL routines',
            reason: 'tlsv1 alert internal error',
            code: 'ERR_SSL_TLSV1_ALERT_INTERNAL_ERROR'
          }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        maxMessageSizeBytes: null,
        maxWriteBatchSize: null,
        maxBsonObjectSize: null,
        primary: null,
        me: null,
        '$clusterTime': null,
        iscryptd: false
      },
      'ac-hqc0sa6-shard-00-02.ep6vddc.mongodb.net:27017' => ServerDescription {
        address: 'ac-hqc0sa6-shard-00-02.ep6vddc.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        minRoundTripTime: 0,
        lastUpdateTime: 424322,
        lastWriteDate: 0,
        error: MongoNetworkError: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
        
            at TLSSocket.<anonymous> (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongodb\lib\cmap\connect.js:286:44)
            at Object.onceWrapper (node:events:633:26)
            at TLSSocket.emit (node:events:518:28)
            at emitErrorNT (node:internal/streams/destroy:170:8)
            at emitErrorCloseNT (node:internal/streams/destroy:129:3)
            at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
          errorLabelSet: Set(1) { 'ResetPool' },
          beforeHandshake: false,
          [cause]: [Error: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
          ] {
            library: 'SSL routines',
            reason: 'tlsv1 alert internal error',
            code: 'ERR_SSL_TLSV1_ALERT_INTERNAL_ERROR'
          }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        maxMessageSizeBytes: null,
        maxWriteBatchSize: null,
        maxBsonObjectSize: null,
        primary: null,
        me: null,
        '$clusterTime': null,
        iscryptd: false
      },
      'ac-hqc0sa6-shard-00-01.ep6vddc.mongodb.net:27017' => ServerDescription {
        address: 'ac-hqc0sa6-shard-00-01.ep6vddc.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        minRoundTripTime: 0,
        lastUpdateTime: 423941,
        lastWriteDate: 0,
        error: MongoNetworkError: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
        
            at TLSSocket.<anonymous> (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongodb\lib\cmap\connect.js:286:44)
            at Object.onceWrapper (node:events:633:26)
            at TLSSocket.emit (node:events:518:28)
            at emitErrorNT (node:internal/streams/destroy:170:8)
            at emitErrorCloseNT (node:internal/streams/destroy:129:3)
            at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
          errorLabelSet: Set(1) { 'ResetPool' },
          beforeHandshake: false,
          [cause]: [Error: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
          ] {
            library: 'SSL routines',
            reason: 'tlsv1 alert internal error',
            code: 'ERR_SSL_TLSV1_ALERT_INTERNAL_ERROR'
          }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        maxMessageSizeBytes: null,
        maxWriteBatchSize: null,
        maxBsonObjectSize: null,
        primary: null,
        me: null,
        '$clusterTime': null,
        iscryptd: false
      }
    },
    stale: false,
    compatible: true,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    setName: 'atlas-10bc9e-shard-0',
    maxElectionId: null,
    maxSetVersion: null,
    commonWireVersion: 0,
    logicalSessionTimeoutMinutes: null
  },
  code: undefined,
  cause: TopologyDescription {
    type: 'ReplicaSetNoPrimary',
    servers: Map(3) {
      'ac-hqc0sa6-shard-00-00.ep6vddc.mongodb.net:27017' => ServerDescription {
        address: 'ac-hqc0sa6-shard-00-00.ep6vddc.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        minRoundTripTime: 0,
        lastUpdateTime: 424325,
        lastWriteDate: 0,
        error: MongoNetworkError: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
        
            at TLSSocket.<anonymous> (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongodb\lib\cmap\connect.js:286:44)
            at Object.onceWrapper (node:events:633:26)
            at TLSSocket.emit (node:events:518:28)
            at emitErrorNT (node:internal/streams/destroy:170:8)
            at emitErrorCloseNT (node:internal/streams/destroy:129:3)
            at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
          errorLabelSet: Set(1) { 'ResetPool' },
          beforeHandshake: false,
          [cause]: [Error: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
          ] {
            library: 'SSL routines',
            reason: 'tlsv1 alert internal error',
            code: 'ERR_SSL_TLSV1_ALERT_INTERNAL_ERROR'
          }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        maxMessageSizeBytes: null,
        maxWriteBatchSize: null,
        maxBsonObjectSize: null,
        primary: null,
        me: null,
        '$clusterTime': null,
        iscryptd: false
      },
      'ac-hqc0sa6-shard-00-02.ep6vddc.mongodb.net:27017' => ServerDescription {
        address: 'ac-hqc0sa6-shard-00-02.ep6vddc.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        minRoundTripTime: 0,
        lastUpdateTime: 424322,
        lastWriteDate: 0,
        error: MongoNetworkError: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
        
            at TLSSocket.<anonymous> (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongodb\lib\cmap\connect.js:286:44)
            at Object.onceWrapper (node:events:633:26)
            at TLSSocket.emit (node:events:518:28)
            at emitErrorNT (node:internal/streams/destroy:170:8)
            at emitErrorCloseNT (node:internal/streams/destroy:129:3)
            at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
          errorLabelSet: Set(1) { 'ResetPool' },
          beforeHandshake: false,
          [cause]: [Error: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
          ] {
            library: 'SSL routines',
            reason: 'tlsv1 alert internal error',
            code: 'ERR_SSL_TLSV1_ALERT_INTERNAL_ERROR'
          }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        maxMessageSizeBytes: null,
        maxWriteBatchSize: null,
        maxBsonObjectSize: null,
        primary: null,
        me: null,
        '$clusterTime': null,
        iscryptd: false
      },
      'ac-hqc0sa6-shard-00-01.ep6vddc.mongodb.net:27017' => ServerDescription {
        address: 'ac-hqc0sa6-shard-00-01.ep6vddc.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        minRoundTripTime: 0,
        lastUpdateTime: 423941,
        lastWriteDate: 0,
        error: MongoNetworkError: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
        
            at TLSSocket.<anonymous> (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongodb\lib\cmap\connect.js:286:44)
            at Object.onceWrapper (node:events:633:26)
            at TLSSocket.emit (node:events:518:28)
            at emitErrorNT (node:internal/streams/destroy:170:8)
            at emitErrorCloseNT (node:internal/streams/destroy:129:3)
            at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
          errorLabelSet: Set(1) { 'ResetPool' },
          beforeHandshake: false,
          [cause]: [Error: 281A0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1605:SSL alert number 80
          ] {
            library: 'SSL routines',
            reason: 'tlsv1 alert internal error',
            code: 'ERR_SSL_TLSV1_ALERT_INTERNAL_ERROR'
          }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        maxMessageSizeBytes: null,
        maxWriteBatchSize: null,
        maxBsonObjectSize: null,
        primary: null,
        me: null,
        '$clusterTime': null,
        iscryptd: false
      }
    },
    stale: false,
    compatible: true,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    setName: 'atlas-10bc9e-shard-0',
    maxElectionId: null,
    maxSetVersion: null,
    commonWireVersion: 0,
    logicalSessionTimeoutMinutes: null
  },
  level: 'error',
  stack: "MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/\n" +
    '    at _handleConnectionErrors (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\connection.js:1165:11)\n' +
    '    at NativeConnection.openUri (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\connection.js:1096:11)',
  timestamp: '2025-08-22T02:15:50.406Z'
}
{
  message: 'SyntaxError: Expected double-quoted property name in JSON at position 88 (line 5 column 1)\n' +
    '    at JSON.parse (<anonymous>)\n' +
    '    at parse (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\body-parser\\lib\\types\\json.js:77:19)\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\body-parser\\lib\\read.js:123:18\n' +
    '    at AsyncResource.runInAsyncScope (node:async_hooks:214:14)\n' +
    '    at invokeCallback (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\raw-body\\index.js:238:16)\n' +
    '    at done (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\raw-body\\index.js:227:7)\n' +
    '    at IncomingMessage.onEnd (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\raw-body\\index.js:287:7)\n' +
    '    at IncomingMessage.emit (node:events:518:28)\n' +
    '    at endReadableNT (node:internal/streams/readable:1698:12)\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)',
  level: 'error',
  service: 'user-service',
  timestamp: '2025-08-22T02:23:31.873Z'
}
{
  service: 'user-service',
  errors: {
    password: ValidatorError: Password must be at least 8 characters long
        at validate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1407:13)
        at SchemaType.doValidate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1391:7)
        at C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\document.js:3111:18
        at process.processTicksAndRejections (node:internal/process/task_queues:85:11) {
      properties: {
        validator: [Function (anonymous)],
        message: 'Password must be at least 8 characters long',
        type: 'minlength',
        minlength: 8,
        path: 'password',
        fullPath: undefined,
        value: 'rishii'
      },
      kind: 'minlength',
      path: 'password',
      value: 'rishii',
      reason: undefined,
      [Symbol(mongoose#validatorError)]: true
    },
    fullName: ValidatorError: Path `fullName` is required.
        at validate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1407:13)
        at SchemaType.doValidate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1391:7)
        at C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\document.js:3111:18
        at process.processTicksAndRejections (node:internal/process/task_queues:85:11) {
      properties: {
        validator: [Function (anonymous)],
        message: 'Path `fullName` is required.',
        type: 'required',
        path: 'fullName',
        fullPath: undefined,
        value: undefined
      },
      kind: 'required',
      path: 'fullName',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose#validatorError)]: true
    }
  },
  _message: 'User validation failed',
  level: 'error',
  message: 'Error during user registration: User validation failed: password: Password must be at least 8 characters long, fullName: Path `fullName` is required.',
  stack: 'ValidationError: User validation failed: password: Password must be at least 8 characters long, fullName: Path `fullName` is required.\n' +
    '    at Document.invalidate (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\document.js:3358:32)\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\document.js:3119:17\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\schemaType.js:1410:9\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:85:11)',
  timestamp: '2025-08-22T02:23:36.309Z'
}
{
  service: 'user-service',
  errors: {
    password: ValidatorError: Password must be at least 8 characters long
        at validate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1407:13)
        at SchemaType.doValidate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1391:7)
        at C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\document.js:3111:18
        at process.processTicksAndRejections (node:internal/process/task_queues:85:11) {
      properties: {
        validator: [Function (anonymous)],
        message: 'Password must be at least 8 characters long',
        type: 'minlength',
        minlength: 8,
        path: 'password',
        fullPath: undefined,
        value: 'rishii'
      },
      kind: 'minlength',
      path: 'password',
      value: 'rishii',
      reason: undefined,
      [Symbol(mongoose#validatorError)]: true
    }
  },
  _message: 'User validation failed',
  level: 'error',
  message: 'Error during user registration: User validation failed: password: Password must be at least 8 characters long',
  stack: 'ValidationError: User validation failed: password: Password must be at least 8 characters long\n' +
    '    at Document.invalidate (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\document.js:3358:32)\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\document.js:3119:17\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\schemaType.js:1410:9\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:85:11)',
  timestamp: '2025-08-22T02:27:07.778Z'
}
{
  service: 'user-service',
  errors: {
    user: ValidatorError: Path `user` is required.
        at validate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1407:13)
        at SchemaType.doValidate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1391:7)
        at C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\document.js:3111:18
        at process.processTicksAndRejections (node:internal/process/task_queues:85:11) {
      properties: {
        validator: [Function (anonymous)],
        message: 'Path `user` is required.',
        type: 'required',
        path: 'user',
        fullPath: undefined,
        value: undefined
      },
      kind: 'required',
      path: 'user',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose#validatorError)]: true
    }
  },
  _message: 'RefreshToken validation failed',
  level: 'error',
  message: 'Error during user registration: RefreshToken validation failed: user: Path `user` is required.',
  stack: 'ValidationError: RefreshToken validation failed: user: Path `user` is required.\n' +
    '    at Document.invalidate (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\document.js:3358:32)\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\document.js:3119:17\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\schemaType.js:1410:9\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:85:11)',
  timestamp: '2025-08-22T02:27:35.553Z'
}
{
  service: 'user-service',
  errors: {
    user: ValidatorError: Path `user` is required.
        at validate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1407:13)
        at SchemaType.doValidate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1391:7)
        at C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\document.js:3111:18
        at process.processTicksAndRejections (node:internal/process/task_queues:85:11) {
      properties: {
        validator: [Function (anonymous)],
        message: 'Path `user` is required.',
        type: 'required',
        path: 'user',
        fullPath: undefined,
        value: undefined
      },
      kind: 'required',
      path: 'user',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose#validatorError)]: true
    }
  },
  _message: 'RefreshToken validation failed',
  level: 'error',
  message: 'Error during user registration: RefreshToken validation failed: user: Path `user` is required.',
  stack: 'ValidationError: RefreshToken validation failed: user: Path `user` is required.\n' +
    '    at Document.invalidate (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\document.js:3358:32)\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\document.js:3119:17\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\schemaType.js:1410:9\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:85:11)',
  timestamp: '2025-08-22T02:31:24.606Z'
}
{
  service: 'user-service',
  errors: {
    user: ValidatorError: Path `user` is required.
        at validate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1407:13)
        at SchemaType.doValidate (C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\schemaType.js:1391:7)
        at C:\Users\Rishi\Desktop\backend\Part -2\social_media_microservices\user-services\node_modules\mongoose\lib\document.js:3111:18
        at process.processTicksAndRejections (node:internal/process/task_queues:85:11) {
      properties: {
        validator: [Function (anonymous)],
        message: 'Path `user` is required.',
        type: 'required',
        path: 'user',
        fullPath: undefined,
        value: undefined
      },
      kind: 'required',
      path: 'user',
      value: undefined,
      reason: undefined,
      [Symbol(mongoose#validatorError)]: true
    }
  },
  _message: 'RefreshToken validation failed',
  level: 'error',
  message: 'Error during user registration: RefreshToken validation failed: user: Path `user` is required.',
  stack: 'ValidationError: RefreshToken validation failed: user: Path `user` is required.\n' +
    '    at Document.invalidate (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\document.js:3358:32)\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\document.js:3119:17\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\mongoose\\lib\\schemaType.js:1410:9\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:85:11)',
  timestamp: '2025-08-22T02:34:50.876Z'
}
{
  service: 'user-service',
  level: 'error',
  message: 'Error during user login: validateLogin is not defined',
  stack: 'ReferenceError: validateLogin is not defined\n' +
    '    at loginUser (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\src\\controllers\\user_controller.js:64:25)\n' +
    '    at Layer.handleRequest (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\router\\lib\\layer.js:152:17)\n' +
    '    at next (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\router\\lib\\route.js:157:13)\n' +
    '    at Route.dispatch (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\router\\lib\\route.js:117:3)\n' +
    '    at handle (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\router\\index.js:435:11)\n' +
    '    at Layer.handleRequest (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\router\\lib\\layer.js:152:17)\n' +
    '    at C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\router\\index.js:295:15\n' +
    '    at processParams (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\router\\index.js:582:12)\n' +
    '    at next (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\router\\index.js:291:5)\n' +
    '    at Function.handle (C:\\Users\\Rishi\\Desktop\\backend\\Part -2\\social_media_microservices\\user-services\\node_modules\\router\\index.js:186:3)',
  timestamp: '2025-08-23T02:56:41.680Z'
}
